backend/api/app/presentation/user/routes.py

```diff
@@ -87,4 +93,159 @@ def verify_ssid():
+                "message": "User information retrieved successfully",
+                "data": result["user"],
+                "status": 200
+            }), 200
+        else:
+            return jsonify({
+                "message": "Userrr not found",
+                "status": 404
+            }), 404
+    except Exception as e:
+        return jsonify({"message": "An unexpected error occurred.", "status": 500}), 500
+    
+@user_bp.route('/change-password', methods=['PATCH'])
+@require_auth
+def change_password():
+    json_data = request.get_json()
+
+    if not json_data:
+        return jsonify({"message": "No data provided.", "status": 400}), 400
+
+    required_fields = ["old_password", "new_password"]
+    missing_fields = [field for field in required_fields if field not in json_data]
+    if missing_fields:
+        return jsonify({
+            "message": f"Missing required fields: {', '.join(missing_fields)}",
+            "status": 400
+        }), 400
+
+    ssid = g.get('auth_token')
+
+    command = ChangePasswordCommand(
+        ssid=ssid,
+        ip_address=request.remote_addr,
+        old_password=json_data.get('old_password'),
+        new_password=json_data.get('new_password')
+    )
+
+    mediator = current_app.config.get('mediator')
+    try:
+        result = mediator.send(command)
+        return jsonify(result), result["status"]
+    except Exception as e:
+        return jsonify({"message": "There was an error during logout.", "error": str(e)}), 500
+    
+@user_bp.route('/delete-account', methods=['POST'])
+@require_auth
+def delete_account():
+    json_data = request.get_json()
+    if not json_data:
+        return jsonify({"message": "No data provided."}), 400
+    
+    password = json_data.get('password')
+    ssid = g.get("auth_token")
+    
+    if not password:
+        return jsonify({"message": "Password is required."}), 400
+
+    try:
+        ses_id = uuid.UUID(ssid)
+        session = SessionRepository.get_active_by_id(ses_id, request.remote_addr)
+        if not session:
+            return jsonify({"message": "Session not found"}), 404
+
+        user_id = session.user_id  
+
+
+        command = DeleteUserCommand(
+            user_id=user_id,
+            password=password
+        )
+
+        mediator = current_app.config.get('mediator')
+        result = mediator.send(command)
+        return jsonify(result), result["status"]
+    except ValueError as e:
+        return jsonify({"message": str(e)}), 400
+    except Exception as ex:
+        return jsonify({"message": "An unexpected error occurred", "error": str(ex)}), 500
+        
+            
+@user_bp.route('/save-general-information', methods=['PUT'])
+@require_auth
+def save_general_info():
+    json_data = request.get_json()
+
+    if not json_data:
+        return jsonify({"message": "No data provided.", "status": 400}), 400
+
+    required_fields = ["name", "lastname", "email", "phone_number", "address", "city", "country"]
```

Comment:
This is too many logic for controller, you should do this using Schemas and should be handled in the Mediator (Handler).
This happened already so check the code for these kind of mistakes.